Client/Client.Qt/Widgets/ChannelListWindow.cpp

```diff
@@ -44,37 +44,13 @@ void ChannelListWindow::updateChannelList()
             std::mutex                   mtx;
             std::unique_lock<std::mutex> lck(mtx);
             mainWidgetStatus.wait(lck);
-
-            int numberOfCoincidences = 0;
-
-            for (int i = 0; i < _channelListMainWindow->count(); ++i)
-            {
-                for (auto it = _channelNames.rbegin(); it != _channelNames.rend(); ++it)
-                {
-                    if (QString::fromStdString(*it) == _channelListMainWindow->item(i)->text())
-                    {
-                        _channelNames.erase(std::remove(_channelNames.begin(), _channelNames.end(), *it), _channelNames.end());
-                    }
-                }
-            }
-            for (auto it = _channelNames.rbegin(); it != _channelNames.rend(); ++it)
+            for (auto &channel : _channelsMap)
             {
-                if (_channelList->count() == 0)
+                if (!channel.second._addToMainChannelsWindow)
                 {
-                    _channelList->addItem(QString::fromStdString(*it));
+                    _channelList->addItem(QString::fromStdString(channel.first));
+                    channel.second._addToMainChannelsWindow = true;
                 }
-                for (int i = 0; i < _channelList->count(); ++i)
```

Comment:
Does this break https://github.com/f1nal3/Juniorgram/issues/217?

Reply:
No